### PR TITLE
Update InfStamina to work with Pal Stamina

### PR DESCRIPTION
**Pull request by ProvidedTimmy**
Makes ResetStamina more consistently reset pal stamina.

### CODE REVIEW

**Comment by Cyk-Fad** [comment]
That's a nice change

File: feature.cpp [hunk 1]
```diff
@@ -404,22 +404,22 @@ void ResetStamina()
...
-		if (!pPalParams)
-			return;
-
-		pPalParams->ResetSP();
+	    if (T[i] != NULL)
+	    {
+	        if (T[i]->IsA(SDK::APalCharacter::StaticClass()))
+	        {
+	            SDK::APalCharacter* Character = (SDK::APalCharacter*)T[i];
+	            if (Character->IsPlayerControlled())
```
Comment (xCENTx) [inline feature.cpp:416]: Is this not just getting the local player character ?

Comment (Tizzi) [reply feature.cpp:416]: no its work for all pals

Comment (xCENTx) [reply feature.cpp:416]: I will do some testing once i get a mount with stamina properties.

Will merge if testing is a success

**Comment by Omega172** [comment]
Cleaner solution in #21

**Comment by xCENTx** [comment]
> Cleaner solution in #21

much cleaner.

Closing this one as I've given the OP enough time to explain their changes and adhere to current coding practices

**Pull request closed without merging**